### ISSUE TITLE
Chronos/automl enhancements before next release

### ISSUE THREAD

**Issue by TheaperDeng** [intel-analytics/BigDL#3184]
_please notice that the list may change according to the developing priority and progress and they are subject to change_

This issue is mainly used for the BigDL 2.0 migration process and avoid unnecessary conflict. @yushan111 

Several major changes is planned to be done before next release (late Oct. 2021), related sub-issues and PRs will be added to track the progress.

1. logger and visualization
This change will mainly affect `zoo/automl/logger`, `zoo/automl/search` and their uts.
    - [x] Change current implementation on `tensorboardx` to `torch.utils.tensorboard`. intel-analytics/analytics-zoo#4620 
    - [ ] Add real time logging and other necessary enhancement. **[low priority target]**
2. Auto model
This change will mainly affect `zoo/chronos/autots/model` and their uts.
    - [x] Add a base auto model to reduce the code duplication intel-analytics/analytics-zoo#4629
    - [x] Add post-fit operation (predict/predict_with_onnx/evaluation/evaluation_with_onnx/save/load/...) intel-analytics/analytics-zoo#4629
    - [x] Add api doc intel-analytics/analytics-zoo#4666 
3. Chronos built-in datasets (@liangs6212 )
This change will mainly affect `zoo/chronos/data/`, use-case and quickstarts
    - [x] add uts intel-analytics/analytics-zoo#4654 
    - [x] use the api in examples/notebooks **[low priority target]**
    - [x] add api doc and user guide intel-analytics/analytics-zoo#4890 
4. documentations and user guide
This change will mainly affect user guides
    - [x] rewrite the autots part in chronos user guide intel-analytics/analytics-zoo#4705
    - [x] add vis documents in user guides intel-analytics/analytics-zoo#4599 
5. Examples on some of our functionalities to help adoptions
This change will mainly affect `zoo/chronos/example`
    - [x] autoprophet intel-analytics/analytics-zoo#4741
    - [x] onnx inference intel-analytics/analytics-zoo#4703
    - [ ] xshardstsdataset **[low priority target]**
    - [x] distributed training/inference intel-analytics/analytics-zoo#4749
6. The integration of doppelganger
    - [x] a clean interface for user to adopt intel-analytics/analytics-zoo#4729
    - [x] ut for doppelganger intel-analytics/analytics-zoo#4850 
    - [x] a good documentation (api docs & user guide) intel-analytics/analytics-zoo#4820 intel-analytics/analytics-zoo#4824 
    - [x] examples (quick start, notebookes) if paper dataset is too long to run, we may need to form a smaller example. intel-analytics/analytics-zoo#4765 
    - [ ] better to have a built-in dataset for user to try **[low priority target]**
7. Model change
    - [x] `LSTMForecaster` support different hidden dim and dropout for different layers intel-analytics/analytics-zoo#4892 
    - [x] TCN support batchnorm or disable weight norm
8. Documentation
    - [x] Make api doc title consistent. https://analytics-zoo.readthedocs.io/en/latest/doc/PythonAPI/Chronos/index.html
    - [x] Proof-read of the user-guide.
    - [x] Proof rerun of the quickstarts.
9. TSDataset
    - [x] TSDataset import tsfresh lazily

**Comment by TheaperDeng** [comment]
> * distributed: Need to manually allocate data to each nodes.
> * tsdata's error message contains extra spaces.
> * built-in dataset: change redownload.

All Done
